src/IxMilia.Dxf/DxfPoint.cs

```diff
@@ -1,94 +1,101 @@
+            return new DxfVector(point.X, point.Y, point.Z);
         }
 
-        public override string ToString()
+        public static bool operator ==(DxfPoint p1, DxfPoint p2)
         {
-            return string.Format("({0},{1},{2})", X, Y, Z);
+            if (object.ReferenceEquals(p1, p2))
+                return true;
+            if (((object)p1 == null) || ((object)p2 == null))
+                return false;
+            return p1.X == p2.X && p1.Y == p2.Y && p1.Z == p2.Z;
         }
 
-        public static DxfPoint Origin
+        public static bool operator !=(DxfPoint p1, DxfPoint p2)
         {
-            get { return new DxfPoint(0, 0, 0); }
+            return !(p1 == p2);
         }
 
-        public static bool operator ==(DxfPoint a, DxfPoint b)
+        public static DxfPoint operator +(DxfPoint p1, DxfVector p2)
         {
-            if (ReferenceEquals(a, b))
-                return true;
-            if (((object)a) == null || ((object)b) == null)
-                return false;
-            return a.X == b.X && a.Y == b.Y && a.Z == b.Z;
+            return new DxfPoint(p1.X + p2.X, p1.Y + p2.Y, p1.Z + p2.Z);
         }
 
-        public static bool operator !=(DxfPoint a, DxfPoint b)
+        public static DxfVector operator -(DxfPoint p1, DxfVector p2)
         {
-            return !(a == b);
+            return new DxfVector(p1.X - p2.X, p1.Y - p2.Y, p1.Z - p2.Z);
         }
 
-        public override int GetHashCode()
+        public static DxfPoint operator *(DxfPoint p, double scalar)
         {
-            return base.GetHashCode();
+            return new DxfPoint(p.X * scalar, p.Y * scalar, p.Z * scalar);
+        }
+
+        public static DxfPoint operator /(DxfPoint p, double scalar)
+        {
+            return new DxfPoint(p.X / scalar, p.Y / scalar, p.Z / scalar);
         }
 
         public override bool Equals(object obj)
         {
+            if (obj == null)
+                return false;
             if (obj is DxfPoint)
+            {
                 return this == (DxfPoint)obj;
+            }
             return false;
         }
-    }
 
-    public class DxfVector : DxfPoint
-    {
-        public DxfVector()
-            : base()
+        public override int GetHashCode()
         {
+            return X.GetHashCode() ^ Y.GetHashCode() ^ Z.GetHashCode();
         }
 
-        public DxfVector(double x, double y, double z)
-            : base(x, y, z)
+        public override string ToString()
         {
+            return string.Format("({0},{1},{2})", X, Y, Z);
         }
 
-        public static DxfVector Zero
+        public static DxfPoint Origin
         {
-            get { return new DxfVector(0, 0, 0); }
+            get { return new DxfPoint(0, 0, 0); }
         }
 
-        public static DxfVector XAxis
+        // the following methods are only used to allow setting individual x/y/z values in the auto-generated readers
+
+        internal DxfPoint UpdateX(double x)
```

Comment:
The name `UpdateX` implies that the "object" is updated, which is not the case. You might ignore the return value and call the method - bug. Some people would call this method `UpdatedX` or `GetUpdatedX` to highlight that the method does not alter the object. Confusing in any case, I prefer constructors only and avoid update methods on immutable objects. However, this is personal oppinion :-)